File: Binary Search/codeforces_379072N.py

# Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/N


import sys


def is_ok(b, y):
    return b * b * b >= y


def solve():
    x = int(sys.stdin.readline())
    for a in range(1, int(x ** (1/3)) + 1):
        y = x - a ** 3
        l, r = 1, 10000

        while l < r:
            mid = (l + r) // 2
            if not is_ok(mid, y):
                l = mid + 1
            else:
                r = mid

        if l ** 3 == y:
            print("YES")
            return
    print("NO")


def main():
    t = int(sys.stdin.readline())
    for _ in range(t):
        solve()


if __name__ == "__main__":
    main()

File: Binary Search/codeforces_379072N.cpp

// Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/N
// submission link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/submission/312742757

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

int isOk(ll b, ll y)
{
    if (b * b * b < y)
        return 0;
    return 1;
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    ll t;
    cin >> t;
    while (t--)
    {
        ll x;
        cin >> x;
        bool got = false;
        for (ll a = 1; a * a * a <= x; a++)
        {
            ll y = x - a * a * a;
            ll l = 1, r = 10000;
            while (l < r)
            {
                ll mid = (l + r) / 2;
                if (isOk(mid, y) == 0)
                    l = mid + 1;
                else
                    r = mid;
            }
            if (l * l * l == y)
            {
                cout << "YES" << endl;
                got = true;
                break;
            }
        }
        if (!got)
            cout << "NO" << endl;
    }
    return 0;
}